### PR TITLE
Test for creating pattern with/without index field

### PR DESCRIPTION
**Pull request by johnreytanquinco**

### CODE REVIEW

File: tests/behat/features/capabilities/custom/post_request_pattern.feature [hunk 1]
```diff
@@ -0,0 +1,193 @@
+Feature: Test REST API POST request
+  In order create a pattern
+  As LIU
```
Comment (magick93) [inline tests/behat/features/capabilities/custom/post_request_pattern.feature:3]: What does LIU mean?

File: tests/behat/features/capabilities/custom/post_request_pattern.feature [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+				}
+			],
+			"field_timeframe": [
+				{
+					"target_id": 13
+				}
+			]
+		}
+    """
+    Then the response status code should be in 422
```
Comment (magick93) [inline tests/behat/features/capabilities/custom/post_request_pattern.feature:193]: Are we able to give a human readable response too?
'422' doesnt tell the API client what they did wrong.

**Review by magick93 — COMMENTED** [review]
The tests look good.
Where is the actual implementation ?

File: tests/behat/features/capabilities/custom/post_request_pattern.feature (hunk 1 above)
Comment (johnreytanquinco) [reply tests/behat/features/capabilities/custom/post_request_pattern.feature:3]: Oh, sorry for the shortcut. It stands for Logged In User.

File: tests/behat/features/capabilities/custom/post_request_pattern.feature (hunk 2 above)
Comment (johnreytanquinco) [reply tests/behat/features/capabilities/custom/post_request_pattern.feature:193]: `422` is unprocessed entity. And its should give a passing result - if the field is empty given it is required. But if you want I can modify the existing method to read like `Then the response status code should be in 422 or unprocessed entity`

Comment (magick93) [reply tests/behat/features/capabilities/custom/post_request_pattern.feature:193]: Ok. But are we able to say way the entity was unprocessed?

**Comment by johnreytanquinco** [comment]
I directly applied the actual implementation in prod site. The reason why, our theme and feature module is behind to the latest site. If you can remember I have added two issues regarding to applying the features I listed. But I havent put priority on it since I was working in some major task/issues. It might take some time to complete this features since I have to test everything in my local first and create behat test for each but I believe its worth it so we can have an updated features applied in the site.

https://gitlab.com/priceinsight/web/priceinsight-features/issues/19
https://gitlab.com/priceinsight/web/priceinsight-features/issues/18

**Pull request merged**